### PR TITLE
add new resources apigee-keystore and references

### PR DESCRIPTION
**Pull request by anuhyapolisetti**
Added support for APIGEE Keystore and References resource

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_apigee_env_keystore
```
```release-note:new-resource
google_apigee_env_references
```

### CODE REVIEW

File: mmv1/products/apigee/api.yaml [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+        description: |
+          Entity tag (ETag) used for optimistic concurrency control as a way to help prevent simultaneous updates from overwriting each other.
+          Used internally during updates.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Enable Synchronizer access':
+          'https://cloud.google.com/apigee/docs/hybrid/v1.8/synchronizer-access#enable-synchronizer-access'
+      api: 'https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations#getsyncauthorization'
+  - !ruby/object:Api::Resource
+    name: 'EnvKeystore'
```
Comment (slevenick) [inline mmv1/products/apigee/api.yaml:706]: Can you change this to the new style of using individual resource YAMLs?

See https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/products/apigee

Comment (anuhyapolisetti) [reply mmv1/products/apigee/api.yaml]: Sure, i have updated it

File: mmv1/products/apigee/EnvKeystore.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        The Apigee environment group associated with the Apigee environment,
+        in the format `organizations/{{org_name}}/environments/{{env_name}}`.
+      required: true
+      input: true
+      url_param_only: true
+  properties:
+    - !ruby/object:Api::Type::String
+      name: 'name'
+      description: |
+        The name of the newly created attachment (output parameter).
```
Comment (slevenick) [inline mmv1/products/apigee/EnvKeystore.yaml:36]: (output parameter) doesn't apply here. This is an input that the user will set

File: mmv1/products/apigee/terraform.yaml [hunk 3]
```diff
@@ -359,6 +359,50 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+      # the resources needed for the acceptance test.
+      name: "apigee_environment_keystore_test"
+      primary_resource_id: "apigee_environment_keystore"
+      test_env_vars:
+        org_id: :ORG_ID
+        billing_account: :BILLING_ACCT
+      skip_docs: true
+      # Resource creation race
+      skip_vcr: true
+    timeouts: !ruby/object:Api::Timeouts
```
Comment (slevenick) [inline mmv1/products/apigee/terraform.yaml:379]: Do we need such large timeouts for these resources? These are needed for resources that take a long time to set up

**Review by slevenick — COMMENTED** [review]
Couple small comments,

File: mmv1/products/apigee/terraform.yaml [hunk 4]
```diff
@@ -359,6 +359,50 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+      # This is a more verbose version of the above that creates all
+      # the resources needed for the acceptance test.
+      name: "apigee_environment_reference_test"
+      primary_resource_id: "apigee_environment_reference"
+      test_env_vars:
+        org_id: :ORG_ID
+        billing_account: :BILLING_ACCT
+      skip_docs: true
+      # Resource creation race
+      skip_vcr: true
```
Comment (slevenick) [inline mmv1/products/apigee/terraform.yaml:400]: Why do we need to skip VCR? I don't think this should apply?

Comment (slevenick) [reply mmv1/products/apigee/terraform.yaml:400]: As long as we ensure correct order with depends_on as needed

File: mmv1/products/apigee/terraform.yaml [hunk 5]
```diff
@@ -359,6 +359,50 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      delete_minutes: 30
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      custom_import: templates/terraform/custom_import/apigee_environment_keystore.go.erb
+  EnvReferences: !ruby/object:Overrides::Terraform::ResourceOverride
+    autogen_async: true
+    import_format: ["{{env_id}}/references/{{name}}", "{{env_id}}/{{name}}"]
+    delete_url: '{{env_id}}/references/{{name}}'
```
Comment (slevenick) [inline mmv1/products/apigee/terraform.yaml:387]: Is this delete_url different from the normal one that will be implied by the resource's base_url?

Comment (anuhyapolisetti) [reply mmv1/products/apigee/terraform.yaml:387]: delete_url is the same base_url and name

File: mmv1/products/apigee/terraform.yaml (hunk 4 above)
Comment (anuhyapolisetti) [reply mmv1/products/apigee/terraform.yaml:400]: Sure, I am removing the skip_vcr

File: mmv1/products/apigee/terraform.yaml (hunk 3 above)
Comment (anuhyapolisetti) [reply mmv1/products/apigee/terraform.yaml:379]: No, resources are created instantly and doesn't have major dependency resources to put major timeouts, I ll go with 1 minute

**Comment by slevenick** [comment]
The tests seem to be failing because the resource can't be found: 
```
    provider_test.go:315: Resource specified by ResourceName couldn't be found: google_apigee_env_keystore.apigee_environment_keystore
--- FAIL: TestAccApigeeEnvKeystore_apigeeEnvironmentKeystoreTestExample (705.80s)
```

Are you able to run these locally?

**Comment by anuhyapolisetti** [comment]
Yes, I am able to run the provider locally. 

As part of this PR, i am creating two new resources "google_apigee_env_keystore" and "google_apigee_env_references".

*google_apigee_env_references* has dependency on the google_apigee_env_keystore.

**Comment by slevenick** [comment]
> Yes, I am able to run the provider locally.
> 
> As part of this PR, i am creating two new resources "google_apigee_env_keystore" and "google_apigee_env_references".
> 
> _google_apigee_env_references_ has dependency on the google_apigee_env_keystore.

Are you able to successfully run the tests? They seem to be failing in our CI. If you're able to run them locally then it *could* be a CI-specific problem, but that would be unusual

**Comment by anuhyapolisetti** [comment]
I have updated the test cases and they ran fine locally, please run the tests again on CI

**Review by slevenick — APPROVED** [review]

**Pull request merged**